skimage/util/_flood_fill.py

```diff
@@ -0,0 +1,186 @@
+"""flood_fill.py - inplace flood fill algorithm
+
+This module provides a function to fill all equal (or within tolerance) values
+connected to a given seed point with a different value.
+"""
+
+import numpy as np
+
+
+def flood_fill(image, seed_point, new_value, *, selem=None, connectivity=None,
+               indices=False, tolerance=None, inplace=False):
+    """Perform flood filling on an image.
+
+    Starting at a specific `seed_point`, connected points equal or within
+    `tolerance` of the seed value are found, then set to `new_value`.
+
+    Parameters
+    ----------
+    image : ndarray
+        An n-dimensional array.
+    seed_point : tuple or int
+        The index into `image` to start filling.
+    new_value : `image` type
+        New value to set the entire fill.  This must be chosen in agreement
+        with the dtype of `image`.
+    selem : ndarray, optional
+        A structuring element used to determine the neighborhood of each
+        evaluated pixel. It must contain only 1's and 0's, have the same number
+        of dimensions as `image`. If not given, all adjacent pixels are
+        considered as part of the neighborhood (fully connected).
+    connectivity : int, optional
+        A number used to determine the neighborhood of each evaluated pixel.
+        Adjacent pixels whose squared distance from the center is larger or
+        equal to `connectivity` are considered neighbors. Ignored if
+        `selem` is not None.
+    indices : bool, optional
+        If True, the output will be an array representing indices of local
+        maxima. If False, the output will be an array of 0's and 1's with the
+        same shape as `image`.
+    tolerance : ``selem`` type, optional
+        If None (default), adjacent values must be strictly equal to the
+        initial value of `image` at `seed_point`.  This is fastest.  If a value
+        is given, a comparison will be done at every point and this tolerance
+        on each side of the initial value will also be filled (inclusive).
+    inplace : bool, optional
+        If True, flood filling is applied to `image` inplace and nothing is
+        returned.  If False, the flood filled result is returned without
+        modifying the input `image`.  Ignored if `indices` is True.
+
+    Returns
+    -------
+    filled : ndarray or tuple[ndarray]
+        If `indices` is false, an array with the same shape as `image` is
+        returned with values equal to (or within tolerance) of the seed point
+        set to `new_value`.  If `indices` is true, a tuple of one-dimensional
+        arrays containing the coordinates (indices) of all found maxima.
+
+    See Also
+    --------
+    skimage.morphology.local_maxima
+    skimage.morphology.local_minima
+    skimage.morphology.h_maxima
+    skimage.morphology.h_minima
```

Comment:
Why are referencing these methods? How are they related to flood-fill? In my opinion `morphology.watershed` has a stronger relation to this.